src/runtime/generator.cpp

```diff
@@ -86,28 +86,34 @@ Context* getReturnContextForGeneratorFrame(void* frame_addr) {
 }
 
 void generatorEntry(BoxedGenerator* g) {
-    assert(g->cls == generator_cls);
-    assert(g->function->cls == function_cls);
+    {
+        STAT_TIMER2(t0, "us_timer_generator_toplevel", g->timer_time);
 
-    threading::pushGenerator(g, g->stack_begin, g->returnContext);
+        assert(g->cls == generator_cls);
+        assert(g->function->cls == function_cls);
 
-    try {
-        RegisterHelper context_registerer(g, __builtin_frame_address(0));
+        threading::pushGenerator(g, g->stack_begin, g->returnContext);
+        try {
+            RegisterHelper context_registerer(g, __builtin_frame_address(0));
 
-        // call body of the generator
-        BoxedFunctionBase* func = g->function;
+            // call body of the generator
+            BoxedFunctionBase* func = g->function;
 
-        Box** args = g->args ? &g->args->elts[0] : nullptr;
-        callCLFunc(func->f, nullptr, func->f->numReceivedArgs(), func->closure, g, func->globals, g->arg1, g->arg2,
-                   g->arg3, args);
-    } catch (ExcInfo e) {
-        // unhandled exception: propagate the exception to the caller
-        g->exception = e;
-    }
+            Box** args = g->args ? &g->args->elts[0] : nullptr;
+            callCLFunc(func->f, nullptr, func->f->numReceivedArgs(), func->closure, g, func->globals, g->arg1, g->arg2,
+                       g->arg3, args);
+        } catch (ExcInfo e) {
+            // unhandled exception: propagate the exception to the caller
+            g->exception = e;
+        }
 
-    // we returned from the body of the generator. next/send/throw will notify the caller
-    g->entryExited = true;
-    threading::popGenerator();
+        // we returned from the body of the generator. next/send/throw will notify the caller
+        g->entryExited = true;
+        threading::popGenerator();
+
+        g->timer_time = getCPUTicks(); // store off the timer that our caller (in g->returnContext) will resume at
```

Comment:
I changed the generator ifdefs from `#if SLOWPATH_TIMERS` to `#if !DISABLE_STATS` since in this particular case we want it to always be there - the timer registered in generatorEntry is the toplevel timer for the generator (much like the main timer in jit.cpp)